runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -77,6 +77,18 @@ protected ILogData getNextEntry(QueuedStreamContext context,
         NavigableSet<Long> getFrom;
         if (context.readCpQueue.size() > 0) {
             getFrom = context.readCpQueue;
+            if (context.readQueue.isEmpty()) {
+                // readQueue is empty, readCpQueue is not.
+                // This is a case where we have had 2 checkpoints
+                // adjacent to each other, and no non-checkpoint
+                // entries in the stream in between the checkpoints
+                // or during the 2nd checkpoint.  Processing of
+                // checkpoint entries will not advance our context
+                // globalPointer, only regular entries in readQueue.
+                // However, we know that readQueue is *empty*, so
+                // we advance globalPointer here.
+                context.globalPointer = maxGlobal;
```

Comment:
This patch concerns me. It seems that there are potentially other cases when a checkpoint will not be followed by updates. So we should set the globalPointer to maxGlobal in those cases as well?

Another corner case would be when there are no updates at all, and there is a checkpoint.